src/V3Coverage.cpp

```diff
@@ -39,14 +39,20 @@ class ExprCoverageEligibleVisitor final : public VNVisitor {
     // STATE
     bool m_eligible = true;
 
+    static const AstNodeDType* getElemDTypep(const AstNodeDType* dtypep) {
+        dtypep = dtypep->skipRefp();
+        while (dtypep && dtypep->virtRefDTypep()) dtypep = dtypep->virtRefDTypep()->skipRefp();
+        return dtypep;
+    }
```

Comment:
I don't think this will work correctly with an associative array.  I think have this return the true/false if eligible something like this (untested)

```suggestion
    static bool elemDTypeEligible(const AstNodeDType* dtypep) {
        dtypep = dtypep->skipRefp();
        if (AstNodeDType* const dtp = dtypep->virtRefDTypep()) {
               if (!elemDTypeEligibleRecurse(dtp)) return false;
        }
        if (AstNodeDType* const dtp = dtypep->virtRefDType2p()) {
               if (!elemDTypeEligibleRecurse(dtp)) return false;
        }
        return !VN_IS(dtypep, ClassRefDType);
    }
```